libgalois/include/katana/DynamicBitset.h

```diff
@@ -86,8 +86,11 @@ class KATANA_EXPORT DynamicBitset {
     KATANA_LOG_DEBUG_ASSERT(
         kNumBitsInUint64 == 64);  // compatibility with other devices
     num_bits_ = n;
+    size_t old_size = bitvec_.size();
     bitvec_.resize((n + kNumBitsInUint64 - 1) / kNumBitsInUint64);
-    reset();
+    if (bitvec_.size() > old_size) {
+      std::fill(bitvec_.begin() + old_size, bitvec_.end(), 0);
+    }
```

Comment:
Just a note that this changes the semantics of `resize` since before the invocation of `reset` cleared everything. I think these semantics for resize make (more) sense, but I'm not sure if there were parts of the code that depended on `resize` also clearing all bits so thought I would call it out.

Reply:
Yes, the callers may need to be fixed if the CI tests fail.